spec/formatting.md

```diff
@@ -685,60 +685,88 @@ and a U+007D RIGHT CURLY BRACKET `}`.
 
 ### Handling Bidirectional Text
 
-_Messages_ contain text which can be bidirectional,
-consisting of a mixture of left-to-right and right-to-left spans of text.
+_Messages_ contain text. Any text can be 
+<a href="https://www.w3.org/TR/i18n-glossary/#dfn-bidirectional-text">bidirectional text</a>.
+That is, the text can can consist of a mixture of left-to-right and right-to-left spans of text.
+The [Unicode Bidirectional Algorithm](http://www.unicode.org/reports/tr9/) [UAX9], aka "UBA",
+is used to determine how bidirectional text is displayed.
 
-When concatenating formatted values,
-the [Unicode Bidirectional Algorithm](http://www.unicode.org/reports/tr9/) [UAX9]
-can produce unexpected or undesirable effects, such as "spillover".
-Formatted values SHOULD be bidirectionally isolated
-so that the directionality of a formatted _expression_
-does not negatively affect the presentation of the overall formatted result.
-
-An implementation MUST define methods for
-determining the directionality of each formatted _expression_.
-The method of determining the directionality of a formatted _expression_
-MAY rely on the introspection of its contents, or on other means.
 The directionality of the message as a whole is provided by the _formatting context_.
 
+When a _message_ is formatted, _placeholders_ are replaced
+with their formatted representation.
+When UBA is applied to the text of a formatted _message_ (including its formatted parts)
+there can be unexpected or undesirable 
+<a href="https://www.w3.org/TR/i18n-glossary/#dfn-spillover-effects">spillover effects</a>.
+Applying <a href="https://www.w3.org/TR/i18n-glossary/#dfn-bidi-isolation">bidi isolation</a>
+to the formatted values helps avoid spillover in a formatted _message_.
+
+The reference data model includes direction for both the _message_ and, separately,
+for each _placeholder_.
```

Comment:
Actually, it doesn't. The data model is about input, and direction is an attribute of the output.

Reply:
Ack. See changes.